jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -1333,6 +1333,7 @@ public void sessionInactivityTimerExpired (Session session)
             if (_sessionIdManager.getSessionHouseKeeper() != null && _sessionIdManager.getSessionHouseKeeper().getIntervalSec() > 0)
             {
                 _candidateSessionIdsForExpiry.add(session.getId());
+                session.stopInactivityTimer(true);
```

Comment:
I think the destroy semantic needs to be more rigorous.  Perhaps just call `_sessionInactivityTimer.destroy()` whenever the session is invalidated (or passivated????)